### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by youmari**
## In this branch I implemented the following : 
- Install and setup devise
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords and stored them in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
```
Comment (dasileker) [inline config/routes.rb:2]: - [x] Nice job on adding the devise however, when the local host start I should see the login page. right now the page is the user page I see. make sure the landing page in the routes is the register or login.

Capture web_4-4-2022_94412_localhost

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @youmari,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- Linters are passing 👌.
- PR is fine 👌.


## Required Changes ♻️
_Check the comments under the review._

**Please add details to the readme file, don't make it the last thing to change.**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (youmari) [reply config/routes.rb:2]: Thanks, @dasileker  for the feedback and Ramadan Mubarak

**Review by dasileker — APPROVED** [review]
### **Hi @youmari,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**